src/sql/src/func.rs

```diff
@@ -4251,12 +4251,24 @@ pub static OP_IMPLS: Lazy<BTreeMap<&'static str, Func>> = Lazy::new(|| {
             params!(ListElementAnyCompatible, ListAnyCompatible) => ElementListConcat => ListAnyCompatible, oid::OP_CONCAT_ELEMENY_LIST_OID;
         },
 
-        //JSON, MAP, RANGE
+        // JSON, MAP, RANGE
         "->" => Scalar {
             params!(Jsonb, Int64) => JsonbGetInt64 { stringify: false } => Jsonb, 3212;
             params!(Jsonb, String) => JsonbGetString { stringify: false } => Jsonb, 3211;
             params!(MapAny, String) => MapGetValue => Any, oid::OP_GET_VALUE_MAP_OID;
-            params!(MapAny, ScalarType::Array(Box::new(ScalarType::String))) => MapGetValues => ArrayAnyCompatible, oid::OP_GET_VALUES_MAP_OID;
+            params!(MapAny, ScalarType::Array(Box::new(ScalarType::String))) => Operation::binary(|ecx, lhs, rhs| {
+                let map_type = ecx.scalar_type(&lhs);
+                let inner_type = map_type.unwrap_map_value_type();
+                if matches!(
+                    inner_type,
+                    ScalarType::List { .. } | ScalarType::Map { .. } | ScalarType::Array(_)
+                ) {
+                    bail_unsupported!(
```

Comment:
There's a `bail_never_supported` which might be more appropriate here if we never plan on supported these types of maps.